concrete/controllers/single_page/dashboard/system/registration/profiles.php

```diff
@@ -24,29 +20,33 @@ public function update_profiles()
         }
 
         if ($this->isPost()) {
-
             $config = $this->getSite()->getConfigRepository();
 
+            $isProfileOptionChanged = $config->get('user.profiles_enabled') === $this->post('public_profiles') ? true : false;
+
             $config->save('user.profiles_enabled', ($this->post('public_profiles') ? true : false));
             $config->save('user.gravatar.enabled', ($this->post('gravatar_fallback') ? true : false));
-            $config->save('user.gravatar.max_level', Loader::helper('security')->sanitizeString($this->post('gravatar_max_level')));
-            $config->save('user.gravatar.image_set', Loader::helper('security')->sanitizeString($this->post('gravatar_image_set')));
+            $config->save('user.gravatar.max_level', $this->app->make('helper/security')->sanitizeString($this->post('gravatar_max_level')));
+            $config->save('user.gravatar.image_set', $this->app->make('helper/security')->sanitizeString($this->post('gravatar_image_set')));
             $config->save('user.display_account_menu', (bool) $this->post('display_account_menu', false));
 
+            if (!$isProfileOptionChanged) {
+                $this->redirect('/dashboard/system/registration/profiles/profiles_updated');
```

Comment:
What about using `$this->flash('success', t('Public profiles settings have been updated.'));` instead of adding a new method?

Reply:
Updated.